File: doc/src/scripts/InfernalRoboticsExample.cs
using System;
using System.Net;
using System.Threading;
using KRPC.Client;
using KRPC.Client.Services.InfernalRobotics;
using KRPC.Client.Services.SpaceCenter;

class InfernalRoboticsExample
{
    public static void Main ()
    {
        using (var connection = new Connection (
            name: "InfernalRobotics Example")) {
            var vessel = connection.SpaceCenter ().ActiveVessel;
            var ir = connection.InfernalRobotics ();

            var group = ir.ServoGroupWithName (vessel, "MyGroup");
            if (group == null) {
                Console.WriteLine ("Group not found");
                return;
            }

            foreach (var servo in group.Servos)
                Console.WriteLine (servo.Name + " " + servo.Position);

            group.MoveRight ();
            Thread.Sleep (1000);
            group.Stop ();
        }
    }
}

File: doc/src/scripts/services/InfernalRoboticsExample.lua
local krpc = require 'krpc'

local conn = krpc.connect('InfernalRobotics Example')
local vessel = conn.space_center.active_vessel

local group = conn.infernal_robotics.servo_group_with_name(vessel, 'MyGroup')
if group == krpc.types.none then
  print('Group not found')
  os.exit(1)
end

for _,servo in ipairs(group.servos) do
  print(servo.name, servo.position)
end

group:move_right()
krpc.platform.sleep(1)
group:stop()
